### PR TITLE
Revert "Change Vec creation to check if gen is lit (and hence needs t…

### PR DESCRIPTION
**Pull request by ucbjrl**
…o be declared)"

This reverts commit dc86e7e1734d6abacb739b488df1de231e6b41b2.
This may address #522 - using chiselCloneType (instead of cloneType) to preserve directionality.

### CODE REVIEW

**Comment by ducky64** [comment]
First problem is that the tests (compile!) aren't passing...

For Aggregate.scala, this doesn't look like a direct (line-by-line) revert, though I think it does match the old behavior.

Do we need to delete the tests, or can we preserve those?

**Comment by ucbjrl** [comment]
I think the fix is simple. Hang on ...

**Comment by donggyukim** [comment]
What's the priority of this PR? It should be merged ASAP for ucb-bar/riscv-boom#20.

**Comment by jackkoenig** [comment]
This should be merged ASAP, It looks good to me, but I don't entirely understand the switch from using SourceInfoTransform to CompileOptionsTransform macro on indexWhere and lastIndexWhere.

Also the removal from the macro on `apply[T <: Data](n: Int, gen: T)`, I'm guessing because these create types so their source locators aren't that important?

@ducky64 can you review?

File: src/test/scala/chiselTests/Vec.scala [hunk 1]
```diff
@@ -32,6 +32,15 @@ class IOTesterMod(vecSize: Int) extends Module {
   io.out := io.in
 }
 
+class OneBitUnitRegVec extends Module {
+  val io = IO(new Bundle {
+    val out = Output(UInt(1.W))
+  })
+  val oneBitUnitRegVec = Reg(Vec(1, 1.U))
```
Comment (ducky64) [inline src/test/scala/chiselTests/Vec.scala]: Should the second Vec argument be a UInt(1.W) instead of a literal?

Comment (ucbjrl) [inline src/test/scala/chiselTests/Vec.scala]: I think @ducky64  is correct. The second `Vec` argument should be a `UInt(1.W)`.

Comment (jackkoenig) [inline src/test/scala/chiselTests/Vec.scala]: Aren't we trying to check Vecs with lit gens here though?

**Pull request merged**